### DISCUSSION TITLE
Mesh Data Layout

### DISCUSSION THREAD

**Discussion by TLBradbury** [jneilliii/OctoPrint-BedLevelVisualizer#458] [Ideas]
Sorry if this has been discussed before, but one of the aspects of this (great) plugin that confuses me the most is that the presentation of the data does not match the physical layout of the bed.

On my print surface, the bottom (closest to me) left corner is 0,0 but on the chart, it is shown in the top left corner. I'm sure you are just taking the Plotly data and presenting it in the same order, but would it be possible to add a toggle to flip the order on the Y-axis so the data (and corrections) are shown in a way that matches their physical locations?

**Comment by jneilliii** [comment]
That's what the flip options are for.

**Reply by jneilliii** [reply]
![image](https://user-images.githubusercontent.com/5249455/129484331-055a1f33-002b-4ab1-9d1d-b50468e5c4b6.png)

![image](https://user-images.githubusercontent.com/5249455/129484351-e0aed605-d4fd-4359-a2f2-e49690927385.png)

**Reply by jneilliii** [reply]
oh, and you have to run an update mesh process using the button after making those changes for it to take effect. It will not re-apply to the current saved data if that option is enabled.

**Comment by TLBradbury** [comment]
Duh.

Thanks!

**Comment by TLBradbury** [comment]
Thanks! I'll get a chance to run it again real soon, since I'm currently printing a new mount for my camera, which of course means I will have to undo the screws for the bed again.

Turns out the reason I didn't see it is the 'Descending y-axis' box is on the Data tab, and I spend almost all of my time on the Corrections tab, but the checkbox impacts both. I also discovered (totally by accident) clicking on one of the grids to be corrected and having the other squares update relative to the new point of reference. Very helpful.

Terry

**Reply by AndrewLindsay** [reply]
This is the sort of information that I was interested in.  How do I work out which position on the chart is my front left corner (the first corner that is probed when I do a bed levelling run).  The silly part of me thought this would be 0,0, it it, or is it Max,Max, or some other variation?

**Reply by TLBradbury** [reply]
Using the flip options to put the values on the chart in their corresponding positions to the print bed is the only way to stay sane with this. IMHO this should be the default layout, but I know nothing about Plotly, so I will refrain from making comments that will demonstrate my ignorance.

Tick the checkbox shown above (Flip Y-Axis) to put the numbers in the correct order. Once they are in the correct positions, it will all make much more sense. jneilliii has done the community a great service with this plugin, and deserves our thanks.

**Reply by jneilliii** [reply]
Thanks for your kind words @TLBradbury. @AndrewLindsay the easiest way to determine if the graph matches real world orientation and determine if you need to use any of the flip options for collection is as follows. Place a business card or something a little thicker in one of the corners of your bed so that it will trigger a probe response earlier than normal. This will generate a bump up on the graph where the business card is and help identify where the plugin thinks the card is based on the returned mesh data. So for example, if you put the card in the front left corner and press the update mesh button the graph will show where it thinks that bump up is. Adjust the flip options accordingly.

**Reply by AndrewLindsay** [reply]
Agreed - I have confirmed the layout in the table and managed to get my bed within 0.05mm of level - astonishing!!!.
I can't get the graph to match up, but I just look at the table and my knob turns are in the single degrees to I'm happy.
Thank your
